### PR TITLE
Add a to_dict method to PdfObjects a from_json method to PdfFile.

### PR DESCRIPTION
**Pull request by SamDepardieu**
This is to allow import/export of the scraped objects to a database in the near future.

### CODE REVIEW

File: pdf_parser/objects/PdfObjects.py [hunk 1]
```diff
@@ -29,6 +31,16 @@ def __init__(self, size, bold, text, page_number, font_face):
...
 
+    def to_dict(self):
+        """Return a dictionary representation of the PdfLine."""
+        return {
+            'size': self.size,
+            'bold': self.bold,
+            'text': self.text,
+            'page_number': self.page_number,
+            'font_face': self.font_face,
+        }
```
Comment (alexwlchan) [inline pdf_parser/objects/PdfObjects.py]: I would suggest [rewriting this class to use attrs](https://glyph.twistedmatrix.com/2016/08/attrs.html), then you get `attr.asdict` for free (along with nice repr, magic methods, and a bunch of other stuff).

Comment (nsorros) [reply pdf_parser/objects/PdfObjects.py]: Mixed feelings about your suggestion, as can be seen in the conversation in hacker new about the post you shared. https://news.ycombinator.com/item?id=12359522 

In general, I dislike too much magic but my problem with attrs I guess is mostly that it adds opinion to the codebase and it seems far from the python way of dealing with this problem. I would avoid it but that is just my opinion :p

Comment (nsorros) [reply pdf_parser/objects/PdfObjects.py]: Mixed feelings about your suggestion, as can be seen in the conversation in hacker new about the post you shared. https://news.ycombinator.com/item?id=12359522 

In general, I dislike too much magic but my problem with attrs I guess is mostly that it adds opinion to the codebase and it seems far from the python way of dealing with this problem. I would avoid it but that is just my opinion

Comment (SoftDrink) [reply pdf_parser/objects/PdfObjects.py]: It's an interesting suggestion. As far as I know, the 3.7 version of Python will include a `dataclass` decorator which intent to do pretty much the same as attrs (see https://www.python.org/dev/peps/pep-0557/), so I guess it won't hurt to go ahead and simplify the class implementation.

Also, I can't see where the "magic" here would impact readability or maintainability, as the attrs lib is easily readable even by someone who never heard of it (in my opinion).

I'll give it a go 👍

Comment (alexwlchan) [reply pdf_parser/objects/PdfObjects.py]: Python 3.7 does indeed include dataclasses, but IMO they're a worse version of attrs made by people who didn't fully understand the decisions made in the original attrs design.

They're also not available until you move to 3.7 everywhere. That may be less of an issue for you, admittedly – but puts it out of reach for a lot of code. Depends on where you're running it; containers you're fine, AWS Lambda you might find harder.

Also members of Python core dev being a bunch of jerks to Hynek (the attrs creator).

</rant>

Comment (SoftDrink) [reply pdf_parser/objects/PdfObjects.py]: I do prefer attrs anyway, I don't like the idea of being strongly dependent of a particular version of Python.

**Comment by SoftDrink** [comment]
Pushed a new version using attrs

**Pull request merged**